stories/FormField-stories.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2025 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import type { Meta, StoryObj } from '@storybook/react';
+import type { ComponentProps } from 'react';
+import { FormField, FormFieldValidation, FormFieldWidth } from '../src/components/form/FormField';
+
+type FormField = typeof FormField;
+
+const meta: Meta<FormField> = {
+  component: FormField,
+  title: 'Echoes/Forms/FormField (internal)',
+  argTypes: {
+    validation: {
+      control: { type: 'select' },
+      options: Object.values(FormFieldValidation),
+      table: {
+        defaultValue: { summary: FormFieldValidation.None },
+      },
+    },
+    width: {
+      control: { type: 'select' },
+      options: Object.values(FormFieldWidth),
+      table: {
+        defaultValue: { summary: FormFieldWidth.Full },
+      },
+    },
+  },
```

Comment:
That's not setup out of the box?

Reply:
If I just set them as args, e.g.

```typescript
args: {
  ...,
  validation: FormFieldValidation.None,
  width: FormFieldWidth.Full,
}
```

Then they appear as text controls

![image](https://github.com/user-attachments/assets/41f60533-5201-4d99-b6b9-512d2c1e1375)

I think we may want to invest more time in general in our storybook documentation. Admittedly I'm not super familiar with it. I was disappointed that storybook does not use my jsdoc comments for documentation.

Reply:
Ah that's super weird, I also have similar typing in TextInput-story and it shows this:
![image](https://github.com/user-attachments/assets/4d975b1f-2458-4485-859a-a7902ba70c18)

But I'm passing the args as string instead of using the enum... That's only difference I noticed.

Reply:
hmm... Yeah storybook seem pretty inconsistent in how it generates controls. Maybe its work updating to see if it has improved?

Reply:
Moving the component definition to the top of the file it fixed it 🤷‍♂️